src/pages/128x64x1/crsfdevice_page.c

```diff
@@ -0,0 +1,221 @@
+/*
+ This project is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Deviation is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Deviation.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "common.h"
+#include "pages.h"
+#include "gui/gui.h"
+
+#ifndef OVERRIDE_PLACEMENT
+enum {
+    LABEL_X          = 2,
+    LABEL_WIDTH      = 120,
+    MSG_X            = 20,
+    MSG_Y            = 10,
+    EDIT_LABEL_WIDTH = 60,
+    EDIT_VALUE_X     = 66,
+    EDIT_VALUE_WIDTH = 56,
+};
+#endif
+
+#if HAS_CRSF_CONFIG
+
+#include "../common/_crsfdevice_page.c"
+
+
+static int folder_rows(int folder) {
+    int count = 0;
+    crsf_param_t *param = crsf_params;
+
+    while (param->id) {
+        if (param->parent == folder && !param->hidden) count += 1;
+        param += 1;
+    }
+    return count;
+}
+
+void button_press(guiObject_t *obj, const void *data)
+{
+    (void)obj;
+    crsf_param_t *param = (crsf_param_t *)data;
+
+    if (param->u.text_sel == param->min_value)
+        param->u.text_sel = param->max_value;
+    else
+        param->u.text_sel = param->min_value;
+
+    CRSF_set_param(param, param->u.text_sel);
+}
+
+static void command_press(guiObject_t *obj, s8 press_type, const void *data)
+{
+    (void)obj;
+
+    if (press_type != -1) {
+        return;
+    }
+
+    crsf_param_t *param = (crsf_param_t *)data;
+
+    if (param->u.status == READY) {
+        CRSF_set_param(param, START);
+    }
+}
+
+static const char *value_textsel(guiObject_t *obj, int dir, void *data)
+{
+    (void)obj;
+    crsf_param_t *param = (crsf_param_t *)data;
+    u8 changed = 0;
+
+    param->u.text_sel = GUI_TextSelectHelper(param->u.text_sel,
+                            param->min_value, param->max_value,
+                            dir, 1, 1, &changed);
+
+    if (changed) CRSF_set_param(param, param->u.text_sel);
+    return current_text(param);
+}
+
+static void press_cb(struct guiObject *obj, s8 press_type, const void *data)
+{
+    (void)obj;
+    if (press_type != -1) {
+        return;
+    }
+
+    crsf_param_t *param = (crsf_param_t *)data;
+
+    switch (param->type) {
+    case FOLDER:
+        current_folder = param->id;
+        show_page(current_folder);
+        break;
+    case UINT8:
+    case UINT16:
+    case FLOAT:
+    case INT8:
+    case INT16:
+    case STRING:
+    case INFO:
+    case OUT_OF_RANGE:
+    default:
+        break;
```

Comment:
will all the other cases get filled? If not, please make sure the compiler optimizes this to a single if statement

Reply:
This is resolved in b863acf1 by eliminating the switch since only one case was used.